tensorflow/lite/micro/testing/micro_test.h

```diff
@@ -149,13 +149,26 @@ extern tflite::ErrorReporter* reporter;
 
 #define TF_LITE_MICRO_EXPECT_NEAR(x, y, epsilon)                               \
   do {                                                                         \
-    auto vx = (x);                                                             \
-    auto vy = (y);                                                             \
+    auto vx = static_cast<float>(x);                                                             \
+    auto vy = static_cast<float>(y);                                                             \
     auto delta = ((vx) > (vy)) ? ((vx) - (vy)) : ((vy) - (vx));                \
-    if (delta > epsilon) {                                                     \
+    if (delta > static_cast<float>(epsilon)) {                                                     \
       micro_test::reporter->Report(                                            \
-          #x " (%f) near " #y " (%f) failed at %s:%d", static_cast<float>(vx), \
-          static_cast<float>(vy), __FILE__, __LINE__);                         \
+          #x " (%f) near " #y " (%f) failed at %s:%d", static_cast<double>(vx), \
+          static_cast<double>(vy), __FILE__, __LINE__);                         \
+      micro_test::did_test_fail = true;                                        \
+    }                                                                          \
+  } while (false)
+
+#define TF_LITE_MICRO_EXPECT_NEAR_RELATIVE(x, y, rel_epsilon)                  \
+  do {                                                                         \
+    auto vx = static_cast<float>(x);                                                             \
```

Comment:
same comment about whether static_cast<float> can be avoided or not. Please let me know if I have missed something.

Reply:
Done